surface/include/pcl/surface/advancing_front_point_type.h

```diff
@@ -0,0 +1,174 @@
+ *
+ * Point Cloud Library (PCL) - www.pointclouds.org
+ * Copyright (c) 2017-, Southwest Research Institute
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ * * Redistributions in binary form must reproduce the above
+ *   copyright notice, this list of conditions and the following
+ *   disclaimer in the documentation and/or other materials provided
+ *   with the distribution.
+ * * Neither the name of Willow Garage, Inc. nor the names of its
+ *   contributors may be used to endorse or promote products derived
+ *   from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * $Id$
+ *
+ */
+
+#ifndef PCL_SURFACE_ADVANCING_FRONT_POINT_TYPE_H_
+#define PCL_SURFACE_ADVANCING_FRONT_POINT_TYPE_H_
+
+#include <pcl/point_types.h>
+#include <pcl/register_point_struct.h>
+#include <pcl/impl/instantiate.hpp>
+#include <ostream>
+
+namespace pcl
+{
+  namespace afront
+  {
+    struct EIGEN_ALIGN16 _AfrontVertexPointType
+    {
+      PCL_ADD_POINT4D;  // This adds the members x,y,z which can also be accessed using the point (which is float[4])
+      PCL_ADD_NORMAL4D; // This adds the member normal[3] which can also be accessed using the point (which is float[4])
+
+      union {
+        struct
+        {
+          float curvature;
+          float max_step;
+          float max_step_search_radius;
+        };
+        float data_c[4];
+      };
+      EIGEN_MAKE_ALIGNED_OPERATOR_NEW;
+    };
+
+    struct AfrontVertexPointType : public _AfrontVertexPointType
+    {
+      inline AfrontVertexPointType (const AfrontVertexPointType &p)
+      {
+        x = p.x;
+        y = p.y;
+        z = p.z;
+        data[3] = 1.0f;
+        normal_x = p.normal_x;
+        normal_y = p.normal_y;
+        normal_z = p.normal_z;
+        data_n[3] = 0.0f;
+        curvature = p.curvature;
+        max_step = p.max_step;
+        max_step_search_radius = p.max_step_search_radius;
+      }
```

Comment:
Why are you required to define this? This looks like the default copy constructor and the compiler should generate one for you automatically.

Reply:
I just noticed that most of the existing if not all did this so I assumed that there was a reason behind it. Let me know if you would like it removed?

Reply:
Yes please. :+1: There's a couple of  things that happen if you define the copy constructor, because intuitively you're telling the compiler "Wait. My copy process is more complex than usual" and the compiler won't generate a couple of useful methods for you in that case. See https://stackoverflow.com/questions/3734247/what-are-all-the-member-functions-created-by-compiler-for-a-class-does-that-hap#3734268 for more info.